frontend/src/bundles/goals/goals-api.ts

```diff
@@ -0,0 +1,47 @@
+import { ApiPath, ContentType } from '~/bundles/common/enums/enums.js';
+import { GoalsApiPath } from '~/bundles/goals/enums/enums.js';
+import { type Http } from '~/framework/http/http.js';
+import { BaseHttpApi } from '~/framework/http-api/http-api.js';
+import { type Storage } from '~/framework/storage/storage.js';
+
+import { type GoalRequestDto, type GoalResponseDto } from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class GoalsApi extends BaseHttpApi {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: ApiPath.GOALS, baseUrl, http, storage });
+    }
+    public async getGoals(): Promise<GoalResponseDto[]> {
```

Comment:
Seems like naming of dto is not following our project conventions

Reply:
I just exported it from shared